sklearn/feature_extraction/text.py

```diff
@@ -1056,18 +1056,19 @@ def transform(self, X, copy=True):
         -------
         vectors : sparse matrix, [n_samples, n_features]
         """
-        if hasattr(X, 'dtype') and np.issubdtype(X.dtype, np.float):
-            # preserve float family dtype
-            X = sp.csr_matrix(X, copy=copy)
-        else:
-            # convert counts or binary occurrences to floats
-            X = sp.csr_matrix(X, dtype=np.float64, copy=copy)
+        X = check_array(X, accept_sparse=["csr"], copy=copy,
```

Comment:
I'm not actually somewhat uncertain of whether this is the right approach. In ``fit`` we always convert to sparse. Maybe we should be consistent? I'm not sure.